### PR TITLE
Add a cross-build check

### PR DESCRIPTION
**Pull request by skitt**
We regularly break Windows builds without realising, e.g. by pulling
in go-iptables in subctl. This adds a job to cross-build  subctl; it
won't detect many failures at the point of origin (in another
project), but it will flag them when the changes affect subctl.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Review by mkolesnik — COMMENTED** [review]
This sounds reasonable, but why not check all cross compile?

**Comment by skitt** [comment]
> This sounds reasonable, but why not check all cross compile?

We could, I wanted to avoid comments about adding too much work in CI ;-).

File: .github/workflows/cross.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+name: Cross-Build Checks
+
+on:
+  pull_request:
+  push:
+    tags:
+      - 'v**'
```
Comment (mkolesnik) [inline .github/workflows/cross.yml]: I think I missed this one, whats the purpose of testing this on tags?

**Comment by mkolesnik** [comment]
> We could, I wanted to avoid comments about adding too much work in CI ;-).

Well given that it's 1 job I think it's reasonable.

If we're worried about clogging up we can change it to have it run only when `ready-to-test` label is applied.

File: .github/workflows/cross.yml (hunk 1 above)
Comment (skitt) [reply .github/workflows/cross.yml]: Copy-pasta...

**Review by dfarrell07 — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Pull request merged**